contracts/Payroll.sol

```diff
@@ -280,50 +278,49 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
-            expected in the equity token. 0% = 0; 1% = 10^16; 100% = 10^18
-     */
-    function determineAllocation(uint256 _denominationTokenAllocation) external employeeMatches {
-        require(_denominationTokenAllocation <= PCT_BASE, ERROR_DENOMINATION_TOKEN_TOO_HIGH);
-
-        uint256 employeeId = employeeIds[msg.sender];
-        Employee storage employee = employees[employeeId];
-
-        employee.denominationTokenAllocation = _denominationTokenAllocation;
-
-        emit DetermineAllocation(employeeId, _denominationTokenAllocation);
-    }
-
     /**
      * @notice Request your `_type == 0 ? 'salary' : _type == 1 ? 'reimbursements' : 'bonus'`
      * @dev Reverts if no payments were made.
      *      Initialization check is implicitly provided by `employeeMatches` as new employees can
      *      only be added via `addEmployee(),` which requires initialization.
      *      As the employee is allowed to call this, we enforce non-reentrancy.
+     * @param _denominationTokenAllocation The percent of payment expected in the denomination token, the rest is
+            expected in the equity token. 0% = 0; 1% = 10^16; 100% = 10^18
      * @param _requestedAmount Requested amount of the denomination token. Must be less than or equal to total owed
-            amount, or zero to request all.
+            amount, or a negative number to request all.
+     * @param _metaData A reference to some details regarding this payment.
      */
-    function payday(uint256 _requestedAmount) external employeeMatches nonReentrant {
+    function payday(uint256 _denominationTokenAllocation, int256 _requestedAmount, string _metaData) external employeeMatches nonReentrant {
+        require(_denominationTokenAllocation <= PCT_BASE, ERROR_DENOMINATION_TOKEN_TOO_HIGH);
+
         uint256 paymentAmount;
         uint256 employeeId = employeeIds[msg.sender];
         Employee storage employee = employees[employeeId];
 
-        // Do internal employee accounting
         // Salary is capped here to avoid reverting at this point if it becomes too big
         // (so employees aren't DDOSed if their salaries get too large)
-        // If we do use a capped value, the employee's lastPayroll date will be adjusted accordingly
+        // If we do use a capped value, the employee's lastPayroll date will still be reset to now
         uint256 totalOwedSalary = _getTotalOwedCappedSalary(employee);
-        paymentAmount = _ensurePaymentAmount(totalOwedSalary, _requestedAmount);
-        _updateEmployeeAccountingBasedOnPaidSalary(employee, paymentAmount);
+        require(totalOwedSalary > 0, ERROR_NO_SALARY);
+
+        if (_requestedAmount >= 0) {
+            paymentAmount = uint256(_requestedAmount);
+        } else {
+            paymentAmount = totalOwedSalary;
+        }
+        require(totalOwedSalary >= paymentAmount, ERROR_INVALID_REQUESTED_AMOUNT);
+
+        // Reset available salary to zero
+        employee.accruedSalary = 0;
```

Comment:
Yes, apart from the instance where they know they should claim less because they haven't worked the full amount. 

On a side note I'd also like to point out the "accruedSalary" is actually salary owed from before the salary has changed. The currently accrued salary since previous `payday()` is determined by the time the individual previously called `payday()` from the line below.